audbackend/core/backend.py

```diff
@@ -290,19 +294,26 @@ def join(
             path joined by :attr:`Backend.sep`
 
         Raises:
-            ValueError: if joined path contains invalid character
+            ValueError: if ``path`` contains invalid character
+                or does not start with ``'/'``,
```

Comment:
As `join()` does not communicate with the backend it feels a little bit strange that you also have to start the first path with `/`. Of cause if you don't do it you cannot use the path with any other function, but still there shouldn't be a reason why you have to start the path with `'/'`.

Reply:
Yes, I was struggling with this decision. But as you said, at the moment there is not use-case for joining with a path that does not start with `/`, so I thought it doesn't hurt to be strict now and maybe soften the constraints at later a point that the other way round.

Reply:
Yes, just wanted to check if you had thought about it ;)

Your solution makes also sense as it is consistent, there is no usage of a path without `/`, and we can remove that restriction when introducing functionality with relative paths.
So let's stay with your decision.